blockly-core/core/ui/block_space/block_space.js

```diff
@@ -817,3 +837,172 @@ Blockly.BlockSpace.prototype.updateScrollableSize = function () {
     this.scrollbarPair.resize();
   }
 };
+
+/**
+ * Establish a mousedown handler on the given dragTarget that will put the
+ * blockspace into a pan-drag mode as long as the mouse is down.
+ * @param {EventTarget} dragTarget - element that will begin pan-drag mode
+ *        when directly clicked.
+ * @param {function} [onDragTargetMouseDown] - optional function called when
+ *        click on the drag target begins (used for hideChaff by BSE)
+ */
+Blockly.BlockSpace.prototype.bindBeginPanDragHandler = function (dragTarget,
+    onDragTargetMouseDown) {
+  this.unbindBeginPanDragHandler();
+  this.panDragData_.target = dragTarget;
```

Comment:
The panDragData object has a fairly rich set of mutable state that goes through several state transitions.  I wonder whether it might be clearer to define a PanDragHandler class that encapsulates the state currently in the panDragData and that has event handler methods that are invoke by the BlockSpace object.

This would help keep the the BlockSpace object from getting too complex, allow for better documentation of the drag data fields and ensure that they are consistently updated.